astropy/nddata/__init__.py

```diff
@@ -48,5 +48,10 @@ class Conf(_config.ConfigNamespace):
         'that data values/uncertainties are not scaled with the unit change.'
     )
 
+    default_ccddata_unit = _config.ConfigItem(
+        'adu',
```

Comment:
This default isn't really backward compatible. I was thinking about setting it to something that is equivalent to undefined for now. We can change the default to `'adu'` in the future.

Reply:
u.dimensionless_unscaled  may be a good "nothing" unit.  This would enable the file to be read without error and the units to be ascertained via inspection of the metadata in the header and set in a subsequent transaction.  

Alternately, CCDData.read could be rewritten to also accept an HDUList so that people can open the file themselves to inspect the header and pass in the open HDUList and proper unit to CCDData.read to do all of the work it currently does.

Reply:
It doesn't look like this is going to move forward in the default distribution on the timeframe I need it, so I have moved forward with a subclass of CCDData that adds the feature.  Others may find it useful, so I attach it here.

[fallback_unit.py.txt](https://github.com/astropy/astropy/files/6291672/fallback_unit.py.txt)